projects/storefrontapp-e2e-cypress/cypress/helpers/b2b/my-company/my-company-form.ts

```diff
@@ -0,0 +1,133 @@
+import { MyCompanyConfig, MyCompanyRowConfig } from './models/index';
+import {
+  loginAsMyCompanyAdmin,
+  scanTablePagesForText,
+} from './my-company.utils';
+
+export function testCreateUpdateFromConfig(config: MyCompanyConfig) {
+  describe(`${config.name} Create / Update`, () => {
+    beforeEach(() => {
+      loginAsMyCompanyAdmin();
+      cy.visit(`${config.baseUrl}`);
+    });
+
+    it(`should create`, () => {
+      cy.get(`cx-organization-list a`).contains('Add').click();
+
+      cy.url().should('contain', `${config.baseUrl}/create`);
+
+      cy.get(`cx-organization-form div.header`).within(() => {
+        cy.get('h3').contains(`Create ${config.name}`, {
+          matchCase: false,
+        });
+      });
+
+      completeForm(config.rows, 'createValue');
+      cy.get('div.header button').contains('Save').click();
+
+      verifyDetails(config, 'createValue');
+
+      cy.get('cx-organization-card cx-icon[type="CLOSE"]').click();
+    });
+
+    it(`should update`, () => {
+      const codeRow = config.rows?.find((row) => row.useInUrl);
+      const nameRow = config.rows?.find((row) => row.sortLabel === 'name');
+
+      cy.wait(3000);
+      scanTablePagesForText(nameRow.createValue, config);
+      cy.get('cx-organization-list a')
+        .contains(`${nameRow.createValue}`)
+        .click({ force: true });
+
+      cy.url().should('contain', `${config.baseUrl}/${codeRow.createValue}`);
+      cy.get(`cx-organization-card a.link`).contains('Edit').click();
+      cy.url().should(
+        'contain',
+        `${config.baseUrl}/${codeRow.createValue}/edit`
+      );
+
+      cy.get(`cx-organization-form`).within(() => {
+        cy.get('div.header').within(() => {
+          cy.get('h3').contains(`Edit ${config.name}`, {
```

Comment:
Not sure if that much nesting is required. Wouldn't one line of .get() do it's job?